client/Containers/FilterSortControl.jsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+import { resetPagination } from '../store/product/productPagination';
+import {
+  setProductFilters,
+  filterByType,
+  filterByStyle,
+  filterByRoom,
+  clearFilters,
+} from '../store/product/productFilters';
+import Button from '../components/styles/Button';
+import buildProductQuery from '../components/utils/buildProductQuery';
+import { fetchProductList } from '../store/product/productList';
+
+// const mapStateToProps = ({ productFilters }) => ({ productFilters });
+
+// const mapDispatchToProps = (dispatch) => ({
+//   filter: (option) => {
+//     const { type, style, room, clear } = option;
+//     if (typeof type === 'string') {
+//       dispatch(filterByType(type));
+//     }
+//     if (typeof style === 'string') {
+//       dispatch(filterByStyle(style));
+//     }
+//     if (typeof room === 'string') {
+//       dispatch(filterByRoom(room));
+//     }
+//     if (clear) {
+//       dispatch(clearFilters());
+//     }
+//     dispatch(resetPagination());
+//   },
+// });
```

Comment:
Originally built this as a class Component, but getting the history required using React hooks, so I commented this out, will delete before merging.